### PR TITLE
format-patch: correctly set UMPF_VERSION with --bb

### PR DESCRIPTION
**Pull request by rohieb**
When exporting with `--bb`, `UMPF_VERSION` is taken from `${STATE}/version`, but this file is never written when we don't set `STABLE` for the "version" branch of `parse()`. Additionally, the version must be parsed back from the input series, otherwise it will always be initialised to "-1", with the result that `UMPF_VERSION` always ends in "-1" even when format-patch'ing an umpf that was built with `-v2` or greater.

Update the test case with an appropriate tag name.

Fixes: 8d0d4ccac5b899526e08 (2024-03-06, "umpf: output version infos to bitbake variables during format-patch")

### CODE REVIEW

File: umpf [hunk 1]
```diff
@@ -928,7 +928,10 @@ parse() {
 		test -n "${content}" || bailout "${cmd} line without value"
 		if ${STABLE}; then
```
Comment (rohieb) [inline umpf:929]: Hmmm. Maybe `if ${STABLE} || [ "${namespace}" = "format_patch" ];` is cleaner code than misusing the `STABLE` variable for this purpose, even if it is only used in that single location for the format-patch use case.

File: umpf [hunk 2]
```diff
@@ -928,7 +928,10 @@ parse() {
...
 			local name="$(<"${STATE}/name")"
-			local version="$(sed "s;${name}/\([0-9]\{8\}\)-.*;\1${VERSION_SEPARATOR}${VERSION};" <<< "${content}")"
+			if [ "${namespace}" = "format_patch" ]; then
+				VERSION="$(sed "s;${name}/[0-9]\{8\}${VERSION_SEPARATOR}\(.*\);\1;" <<< "${content}")"
```
Comment (rohieb) [inline umpf:932]: Note that this also makes it impossible to get something like `20240314-4` or similar from a 20240314-1 tag by calling `umpf format-patch -v4`.

File: umpf [hunk 3]
```diff
@@ -928,7 +928,10 @@ parse() {
...
+			fi
+			local version="$(sed "s;${name}/\([0-9]\{8\}\)${VERSION_SEPARATOR}.*;\1${VERSION_SEPARATOR}${VERSION};" <<< "${content}")"
```
Comment (michaelolbrich) [inline umpf:934]: That's the wrong place to do this. Try this:
```
format_patch_version() {
       local name="$(<"${STATE}/name")"
       local version="${content#${name}/}"
       echo "${version}" > "${STATE}/version"
       echo "${content}" > "${STATE}/tag"
       echo "$line" >&${series_out}
[...]
```

Comment (rohieb) [reply umpf:931]: Thanks, that seems to work.

**Review by michaelolbrich — APPROVED** [review]
Looks good now. Thanks!

**Pull request merged**